From c3603b0a6b04477880352a3c38ddef07d15f06bf Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 13 Mar 2024 14:19:25 +0100
Subject: [PATCH 0778/1695] dts: stm32: disable IWDG1 and IWDG2 during low
 power on stm32mp15x-dkx boards

Update shadow value of otp HW2 to disable IWDG1 and IWDG2 during standby
and stop on stm32mp15x-dkx boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d057e11f23365a622d732bc799704b62e57cab9f dts: stm32: disable IWDG1 and IWDG2 during low power on stm32mp15x-dkx boards]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366596
Change-Id: Ia08e270b8dd88ce8cc375008591aa6f48dc807cf
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372976
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 76649c2c0..7c3dbe2da 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/power/stm32mp1-power.h>
 #include <dt-bindings/mfd/st,stpmic1.h>
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/soc/stm32mp15-etzpc.h>
 #include <dt-bindings/tamper/st,stm32-tamp.h>
 
@@ -86,6 +87,15 @@
 		};
 	};
 
+	shadow-prov {
+		compatible = "st,provisioning";
+		hw2_prov {
+			nvmem-cells = <&hw2_otp>;
+			st,shadow-value = <0x000001E0>;
+			st,shadow-lock = <STICKY_LOCK_SR>;
+		};
+	};
+
 	sound {
 		compatible = "audio-graph-card";
 		label = "STM32MP1-DK";
-- 
2.47.2

